src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs

```diff
@@ -25,8 +26,18 @@ public static int Main(string[] args)
                 CompilerServerLogger.LogException(ex, "Error loading application settings");
             }
 
-            var controller = new DesktopBuildServerController(appSettings);
-            return controller.Run(args);
+            try
+            {
+                var controller = new DesktopBuildServerController(appSettings);
+                return controller.Run(args);
+            }
+            catch (TypeInitializationException ex) when (ex.InnerException is FileNotFoundException)
+            {
+                // Assume FileNotFoundException was the result of a missing
+                // compiler assembly. Log the exception and terminate the process.
+                CompilerServerLogger.LogException(ex, "File not found");
+                return CommonCompiler.Failed;
```

Comment:
Yes, this will kill the server but this should only get hit when there are assemblies required during start up of the server, not when the compiler for an individual request fails to load.